spel/scripts/amigen7-build.sh

```diff
@@ -173,8 +186,33 @@ bash -x "${ELBUILD}"/CleanChroot.sh
 echo "Executing PreRelabel.sh"
 bash -x "${ELBUILD}"/PreRelabel.sh
 
-echo "Saving the aws cli version to the manifest"
-(chroot "${CHROOT}" /usr/bin/aws --version) > /tmp/manifest.log 2>&1
+if [[ "${SPEL_CLOUDPROVIDER}" == "azure" ]]
+then
+    echo "Configuring waagent"
+    #per https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-centos#centos-70
+    chroot "${CHROOT}" /usr/sbin/chkconfig waagent on
+    chroot "${CHROOT}" /usr/bin/sed -i 's|USERCTL="yes"|USERCTL="no"|' /etc/sysconfig/network-scripts/ifcfg-eth0
+    echo 'NM_CONTROLLED="no"' >> "${CHROOT}"/etc/sysconfig/network-scripts/ifcfg-eth0
+    chroot "${CHROOT}" /usr/bin/sed -i 's|DHCP_HOSTNAME=localhost.localdomain||' /etc/sysconfig/network-scripts/ifcfg-eth0
+    ## Not per above ref, added to allow cloud-init or user to manage resource disk to match Azure Marketplace Ubuntu image settings
+    chroot "${CHROOT}" /usr/bin/sed -i 's|ResourceDisk.Format=y|ResourceDisk.Format=n|' /etc/waagent.conf
+    ## End Not per above ref, added to allow cloud-init or user to manage resource disk to match Azure Marketplace Ubuntu image settings
+    chroot "${CHROOT}" /usr/bin/sed -i 's|Provisioning.Enabled=y|Provisioning.Enabled=n|' /etc/waagent.conf
+    chroot "${CHROOT}" /usr/bin/sed -i 's|Provisioning.UseCloudInit=n|Provisioning.UseCloudInit=y|' /etc/waagent.conf
+    chroot "${CHROOT}" /usr/sbin/waagent -force -deprovision
+fi
+
+if [[ "${SPEL_CLOUDPROVIDER}" == "aws" ]]
+then
+    echo "Saving the aws cli version to the manifest"
+    (chroot "${CHROOT}" /usr/bin/aws --version) > /tmp/manifest.log 2>&1
+fi
+
+if [[ "${SPEL_CLOUDPROVIDER}" == "azure" ]]
```

Comment:
Can we combine this with the above `if` block, using `elif` instead of a wholly new `if` block?

Reply:
...or a `case`/`esac` block (esp. if ever likely to add things for other services like Google Compute)